typescript/layout.ts

```diff
@@ -0,0 +1,115 @@
+import { Image } from "./action";
+import { Component } from "../typescript/component";
+
+type Object = {
```

Comment:
I think its better not to use Object as type. It will create conflict with the native Object.
Rename it to something more relavant